client/block-builder/src/lib.rs

```diff
@@ -58,298 +58,308 @@ use sc_client_api::backend;
+    /// output of this block builder without having access to the full storage.
+    pub fn new(
+        api: &'a A,
+        parent_hash: Block::Hash,
+        parent_number: NumberFor<Block>,
+        record_proof: RecordProof,
+        inherent_digests: DigestFor<Block>,
+        backend: &'a B,
+    ) -> Result<Self, ApiErrorFor<A, Block>> {
+        let header = <<Block as BlockT>::Header as HeaderT>::new(
+            parent_number + One::one(),
+            Default::default(),
+            Default::default(),
+            parent_hash,
+            inherent_digests,
+        );
+
+        let mut api = api.runtime_api();
+
+        if record_proof.yes() {
+            api.record_proof();
+        }
+
+        let block_id = BlockId::Hash(parent_hash);
+
+        info!("Api call to initialize the block");
+        api.initialize_block_with_context(&block_id, ExecutionContext::BlockConstruction, &header)?;
+
+        Ok(Self {
+            parent_hash,
+            extrinsics: Vec::new(),
+            api,
+            block_id,
+            backend,
+        })
+    }
+
+    /// Push onto the block's list of extrinsics.
+    ///
+    /// This will ensure the extrinsic can be validly executed (by executing it).
+    pub fn push(&mut self, xt: <Block as BlockT>::Extrinsic) -> Result<(), ApiErrorFor<A, Block>> {
+        // let block_id = &self.block_id;
+        // let extrinsics = &mut self.extrinsics;
+
+        //FIXME add test execution with state rejection
+        info!("Pushing transactions without execution");
+        self.extrinsics.push(xt);
+        Ok(())
+
+        // info!("Going to call api tx execution");
+        // self.api.execute_in_transaction(|api| {
+        // 	match api.apply_extrinsic_with_context(
+        // 		block_id,
+        // 		ExecutionContext::BlockConstruction,
+        // 		xt.clone(),
+        // 	) {
+        // 		Ok(Ok(_)) => {
+        // 			extrinsics.push(xt);
+        // 			TransactionOutcome::Commit(Ok(()))
+        // 		}
+        // 		Ok(Err(tx_validity)) => {
+        // 			TransactionOutcome::Rollback(
+        // 				Err(ApplyExtrinsicFailed::Validity(tx_validity).into()),
+        // 			)
+        // 		},
+        // 		Err(e) => TransactionOutcome::Rollback(Err(e)),
+        // 	}
+        // })
+    }
+
+    /// Consume the builder to build a valid `Block` containing all pushed extrinsics.
+    ///
+    /// Returns the build `Block`, the changes to the storage and an optional `StorageProof`
+    /// supplied by `self.api`, combined as [`BuiltBlock`].
+    /// The storage proof will be `Some(_)` when proof recording was enabled.
+    pub fn build(
+        mut self,
+    ) -> Result<BuiltBlock<Block, backend::StateBackendFor<B, Block>>, ApiErrorFor<A, Block>> {
+        let block_id = &self.block_id;
+
+        let extrinsics = self.extrinsics.clone();
+        let parent_hash = self.parent_hash;
+        let extrinsics_hash = BlakeTwo256::hash(&extrinsics.encode());
+
+        //FIXME
```

Comment:
From my point of view, the following code is fixed now, so //fixme comment is no longer necessary to keep